ql/src/java/org/apache/hadoop/hive/ql/io/sarg/ConvertAstToSearchArg.java

```diff
@@ -106,6 +105,14 @@ private static enum BoxType {
     parse(expression);
   }
 
+  /**
+   * Returns whether the given expression can be completely converted to a search argument.
+   * @return True if the expression can be converted completely, otherwise false.
+   */
+  public boolean isCompleteConversion() {
```

Comment:
could it be package-private?

Reply:
No this is used for checking whether the hive filter can be completely converted into a search argument.

Reply:
but you refer to it in the same class

Reply:
Not anymore :)